java-frontend/src/main/java/org/sonar/java/se/checks/BooleanGratuitousExpressionsCheck.java

```diff
@@ -46,6 +46,7 @@ public void checkEndOfExecution(CheckerContext context) {
   private void reportBooleanExpression(CheckerContext context, AlwaysTrueOrFalseExpressionCollector atof, Tree condition, boolean isTrue) {
     if (!AlwaysTrueOrFalseExpressionCollector.hasUnreachableCode(condition, isTrue)) {
       Set<List<JavaFileScannerContext.Location>> flows = atof.flowForExpression(condition).stream()
+        .filter(flow -> !flow.isEmpty())
```

Comment:
Why not rather filter out empty flows before returning in `flowForExpression` ?!